packages/blade/src/components/Collapsible/CollapsibleBodyContent.native.tsx

```diff
@@ -1,13 +1,121 @@
+import { View } from 'react-native';
+import type { CollapsibleBodyContentProps } from './types';
+import { useCollapsible } from './CollapsibleContext';
+import {
+  getCollapsibleBodyContentBoxProps,
+  getOpacity,
+  getTransitionDuration,
+  getTransitionEasing,
+} from './commonStyles';
+import { nativeStyles } from './styles.native';
+import { Box } from '~components/Box';
+import { useTheme } from '~components/BladeProvider';
+import { castNativeType } from '~utils';
 
-// TODO: implement
-type CollapsibleBodyContentProps = {
-  children: ReactNode;
+type AnimatedStyledCollapsibleBodyContentProps = {
+  isExpanded: boolean;
 };
 
-const CollapsibleBodyContent = ({ children }: CollapsibleBodyContentProps): ReactElement => (
-  <Text>To implement {children}</Text>
-);
+const AnimatedStyledCollapsibleBodyContent = styled(
+  Animated.View,
+)<AnimatedStyledCollapsibleBodyContentProps>(() => {
+  return {
+    overflow: 'hidden',
+  };
+});
+
+const CollapsibleBodyContent = ({ children }: CollapsibleBodyContentProps): ReactElement => {
+  const { isExpanded, direction } = useCollapsible();
+  const { theme } = useTheme();
+
+  const opacity = useSharedValue(getOpacity({ isExpanded }));
+
+  // `undefined` implies no height restrictions which is analogous to `auto` on web
+  const height = useSharedValue(isExpanded ? undefined : 0);
+  const [layoutHeight, setLayoutHeight] = useState(0);
+
+  // Keeps track of running animation to control absolute / relative positioning
+  const [isAnimating, setIsAnimating] = useState(false);
+  const onAnimationComplete = useCallback((): void => setIsAnimating(false), []);
+
+  const duration = castNativeType(getTransitionDuration(theme));
+  const easing = castNativeType(getTransitionEasing(theme));
+
+  useEffect(() => {
+    setIsAnimating(true);
+
+    opacity.value = withTiming(getOpacity({ isExpanded }), { duration, easing });
+    height.value = withTiming(
+      // Animates the height to the measured value
+      isExpanded && layoutHeight ? layoutHeight : 0,
+      { duration, easing },
+      (isComplete) => {
+        // Only run this if the animation ran uninterrupted, for eg collapsing the content before it expanded fully
+        if (isComplete) {
+          // The callback `onAnimationComplete` has to be declared outside this, on JS thread
+          runOnJS(onAnimationComplete)();
+        }
+      },
+    );
+  }, [isExpanded, opacity, duration, easing, height, layoutHeight, onAnimationComplete]);
+
+  const animatedStyles = useAnimatedStyle(() => {
+    return {
+      opacity: opacity.value,
+      height: height.value,
+    };
+  });
+
+  /**
+   * Tracks the height of content so we can animate height to and from 0 to the content's height.
+   * **Note:** We can't animate height from 0 to auto or vice-versa.
+   */
+  const onLayout: (event: LayoutChangeEvent) => void = useCallback(
+    (event) => {
+      if (event.nativeEvent.layout.height > layoutHeight) {
```

Comment:
Do we need this condition? Like onLayout will always run each time a layout change happens either when height increases or decreases. 

If we put this condition and imagine inside an accordion there is some sort of UI which can be removed or dismissed then the height should decrease but we will skip saving the height.

Reply:
Right, had left figuring this part out for Accordion implementation, modified the logic correctly in that.